doc/Manuals/ug-syspar_x-road_v6_system_parameters.md

```diff
@@ -531,3 +535,46 @@ TLS_DHE_RSA_WITH_AES_256_CBC_SHA256
+
+### 6.1 The configuration file
+
+The file `/etc/xroad/services/local.properties` replaces the file `/etc/xroad/services/local.conf` starting from X-Road version 7. 
+
+The `local.properties` file's contents consists of key-value pairs such as `XROAD_SIGNER_PARAMS=-XX:MaxMetaspaceSize=100m`. Notice that you cannot use quotes in the values. If you have multiple values for one key, use space to separate the values e.g. `XROAD_SIGNER_PARAMS=-Xmx100m -XX:MaxMetaspaceSize=100m`. Also notice that with `local.properties` you can only add new parameters and override old ones – you cannot remove existing default parameters.
+
+Example of `/etc/xroad/services/local.properties` with modifications that override the default Java memory parameters for Security Server:
+
+```
+XROAD_PROXY_PARAMS=-Xms200m -Xmx1024m
+```
+
+Example of `/etc/xroad/services/local.properties` with modifications that override the default Java memory parameters for Central Server:
+
+```
+XROAD_JETTY_PARAMS=-Xms200m -Xmx1024m
+```
+
+All possible properties to adjust in this file:
+
+| Property | Details |
+|--------------------|---------|
+| XROAD_SIGNER_PARAMS | Parameters for the Signer |
+| XROAD_ADDON_PARAMS | Parameters for all addons |
+| XROAD_CONFCLIENT_PARAMS | Parameters for the Configuration Client |
+| XROAD_CONFPROXY_PARAMS | Parameters for the Configuration Proxy |
+| XROAD_JETTY_PARAMS | Parameters for the Central Server |
+| XROAD_MONITOR_PARAMS | Parameters for the Environmental Monitor |
+| XROAD_OPMON_PARAMS | Parameters for the Operational Monitor |
+| XROAD_PROXY_PARAMS | Parameters for the Security Server |
+| XROAD_PROXY_UI_API_PARAMS | Parameters for the Security Server's admin API |
+| XROAD_SIGNER_CONSOLE_PARAMS | Parameters for the Signer Console |
+
+### 6.2 Legacy configuration file support in X-Road v7
+
+If you have upgraded from X-Road version 6 to version 7, you might still be using `local.conf` script file with legacy parameters (non *XROAD_* prefixed parameters), which is considered deprecated in X-Road version 7. The legacy file `local.conf` **will not** be included in the configuration backups anymore – the new file `local.properties` will be included instead, therefore it is strongly recommended migrating to the new `local.properties` file.
+
+The legacy file `local.conf` will still be supported along with the legacy parameters. It is possible for both files to exist side-by-side. In this case the `local.conf` file gets executed as a script and the values from `local.properties` are appended afterwards. Notice that if your `local.conf` script contains new *XROAD_* prefixed parameters and old legacy parameters, the new *XROAD_* prefixed parameters will always be preferred.
```

Comment:
I believe local.properties is applied first, and local.conf can see (and modify) those.

Reply:
Yeah good catch – I remembered the order wrong from some discussion